site/blog/harder-better-prompter-stronger.md

```diff
@@ -0,0 +1,345 @@
+
+<!-- truncate -->
+
+Let's dig into system prompts while secretly lamenting someone's unfortunate choice of phrasing for this technique.
+
+## The rise of system prompt hardening
+
+The bottom line is we care because we want system prompts to remain both intact and as intended. If they change, it should be because we designed things that way.
+
+When we don't care about security, we suffer a similar misfortune to the influx of non-developer vibe coders:
+
+<div style={{textAlign: 'center'}}>
+
+![Twitter post about vibe coders](/img/blog/harder-better-prompter-stronger/twitter_post_about_vibe_coders.png)
+
+</div>
+
+There are numerous security issues we encounter around system prompts that manipulators create, as our friend above has discovered, for "fun" or personal gain:
+
+- Prompt injection
+- Cache exploitation
+- Instruction overriding
+- System prompt leakage
+- Cross-context injection
+- Tool use manipulation
+- ... aaaaand those are just a few!
+
+The security of AI models, and LLM reliability, depends on how well we fend off these attacks. They are constantly evolving. [Imprompter](https://imprompter.ai/) is a great example - it doesn't take long for adversaries to develop new techniques.
+
+## Why system prompt hardening matters
+
+We (unwillingly) accept that users will attempt to mess with AI systems. Let's reveal (_insert cape flourish here_) some of the consequences:
+
+- The LLM tone drifts
+- Roleplaying as other personas
+- Safety filters ignored
+- Reverse engineering for prompt injection refinement
+- Impersonated (brand, users - anything)
+- Exposing sensitive data
+- Enabling developer-only modes (!!)
+
+## Techniques to harden system prompts
+
+There are multiple strategies to harden system prompts. To name a few:
+
+1. **Instruction shielding**: Prevent the model from accepting new instructions that override the original intended behaviour.
+2. **Syntax reinforcement**: Make a prompt's structure harder to manipulate.
+3. **Layered prompting**: Use multiple models or prompt stages.
+
+Ultimately, we should have a pipeline for our prompts consisting of tools. Most of the tooling and techniques (like guards, prompt chaining, etc.) operate outside the LLM; prompt hardening happens at the LLM level.
+
+## How to harden a system prompt
+
+Imagine we've started building our prompt pipeline. We're excited to send out our app into the wild. It's a tad uncomfortable because we don't have anything to confirm our system prompt's strength - so it's time for evaluations (evals).
+
+We can do this with Promptfoo (I work there - _>what a coincidence_). It's open source, so you can get started yourself. I assure you that I have not accidentally uploaded any API keys, so you'd need your own.
+
+```yaml title="promptfooconfig.yaml"
+prompts:
+  - 'Convert this English to {{language}}: {{input}}'
```

Comment:
at some point we should introduce a system prompt or make this more about prompt hardening in general and less about system prompt hardening